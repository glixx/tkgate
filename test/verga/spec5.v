module top;
  integer i;
  reg  a,b;
  wire x,y;

  initial
    $monitor("%4d: a=%d b=%d - x=%d y=%d",$time,a,b,x,y);

  initial
    for (i = 0;i < 4;i = i + 1)
      begin
	{a,b} = i;
	#100;
      end

  foo f(a,b,x,y);

endmodule

module foo #(.p1(10), .p2(21)) (a,b,x,y);
input a,b;
output x,y;

  specify
    if (a) (a *> x) = p1;
    if (!a) (a *> x) = p2;
    (b *> x) = 12;
    (a, b *> y) = 17;
  endspecify

  assign {x,y} = a + b;

endmodule
