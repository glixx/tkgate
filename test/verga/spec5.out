   0: a=0 b=0 - x=x y=x
  17: a=0 b=0 - x=x y=0
  21: a=0 b=0 - x=0 y=0
 100: a=0 b=1 - x=0 y=0
 117: a=0 b=1 - x=0 y=1
 200: a=1 b=0 - x=0 y=1
 210: a=1 b=0 - x=1 y=1
 212: a=1 b=0 - x=0 y=1
 300: a=1 b=1 - x=0 y=1
 312: a=1 b=1 - x=1 y=1
 317: a=1 b=1 - x=1 y=0
